ndcube/utils/wcs.py

```diff
@@ -197,18 +202,37 @@ def _wcs_slicer(wcs, missing_axis, item):
                 item_checked.append(slice(0, 1))
         # if all are slice in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
             for i, it in enumerate(item_checked):
                 if isinstance(it, int):
                     missing_axis[i] = True
+    else:
+        print("TypeError: Please look inside the code block and see what went wrong and fix.")
```

Comment:
This should explicitly raise a ```TypeError```.  To do this, write
```python
raise TypeError("item type is {0}.  Must be int, slice, or tuple of ints and/or slices.".format(type(item))
```
(The ```{0}``` and ```.format()``` at the end of the string insert the result of ```type(item)``` into the string.)

Reply:
Yup, done.